src/robotFactory.js

```diff
@@ -1,10 +1,83 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {
+    x: 0, y: 0, z: 0,
+  }, chipVersion) {
```

Comment:
You simply assign a new object with all the fields. If you are given an empty object as class arguments, it will override your default object.
Why are you setting the z coordinate in BaseRobot? Read the conditions of the task carefully

```suggestion
  constructor(name, weight, coords, chipVersion) {
```